src/util/Platform.ts

```diff
@@ -20,3 +21,61 @@ export const currentPlatform: Platform | null = (() => {
             return null;
     }
 })();
+
+export enum UbuntuVersion {
+    v14_04 = "14.04", // End of Life: April 2024
+    v16_04 = "16.04", // End of Life: April 2026
+    v18_04 = "18.04", // End of Life: April 2028
+    v20_04 = "20.04", // End of Life: April 2030
+    v20_10 = "20.10", // Dead
+    v21_04 = "21.04", // Dead
+    v21_10 = "21.10", // Dead
+    v22_04 = "22.04", // End of Life: April 2032
+    v22_10 = "22.10", // End of Standard Support: July 20, 2023
+    v23_04 = "23.04", // End of Life: ???
+    v23_10 = "23.10", // End of Life: ???
+    Unknown = "unknown",
+}
+
+export async function readLinuxReleaseDetails(): Promise<Record<string, string> | null> {
+    let data: string;
+    try {
+        data = await fs.readFile("/etc/os-release", "utf8");
+    } catch (err) {
+        console.log(`Failed to read /etc/os-release: ${err}`);
+        return null;
+    }
+    const lines: string[] = data.split("\n");
+    const releaseDetails: Record<string, string> = {};
+    lines.forEach((line) => {
+        // Split the line into an array of words delimited by "=" and remove quotes
+        const words = line.split("=");
+        if (words.length > 1) {
+            releaseDetails[words[0].trim().toLowerCase()] = words[1].trim().replace(/^"(.*)"$/, '$1');
+        }
+    });
+    return releaseDetails
+}
+
+export async function readUbuntuVersion(): Promise<UbuntuVersion | null> {
```

Comment:
I would add a short documentation mentioning that `null` is returned if the OS does not seem to be Ubuntu